beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -309,10 +309,9 @@ func (s *Service) isProcessedBlock(ctx context.Context, blk block.SignedBeaconBl
 	if blk.Block().Slot() <= finalizedSlot {
 		return true
 	}
-	blockExistsInDB := s.cfg.DB.HasBlock(ctx, blkRoot) || s.cfg.Chain.HasInitSyncBlock(blkRoot)
 	// If block exists in our db and is before or equal to our current head
 	// we ignore it.
-	if blockExistsInDB && s.cfg.Chain.HeadSlot() >= blk.Block().Slot() {
+	if s.cfg.Chain.HasBlock(ctx, blkRoot) && s.cfg.Chain.HeadSlot() >= blk.Block().Slot() {
```

Comment:
```suggestion
	if s.cfg.Chain.HeadSlot() >= blk.Block().Slot() && s.cfg.Chain.HasBlock(ctx, blkRoot) {
```

Flip the order of operations here to avoid a db lookup when the first condition is false.